### ISSUE TITLE
Deleting children in admin inline breaks the structure

### ISSUE THREAD

**Issue by olivierdalang** [django-mptt/django-mptt#576]
Hi !

I have a MPTT tree and I have an inline form allowing to edit direct children in django admin.

With this simple setup, **deleting inlines instances from the parent form breaks the MPTT tree**.
The strange thing is that the delete() method of the instances is called (so it's not the same problem as having a queryset.delete() which doesn't individually delete items).

I just tested this on a blank project on django 1.11.2 and django-mptt 0.8.7

```
class MyModel(MPTTModel):
    parent = models.ForeignKey('MyModel', blank=True, null=True)
    ...
```
and the following admin
```
class MyModelInline(admin.TabularInline):
    model = MyModel
class MyModelAdmin(admin.ModelAdmin):
    inlines = [MyModelInline,]
    ...
```

(note that issue #233 has a similar name but is probably a different issue)

**Comment by olivierdalang** [comment]
Ok I inspected a bit more an could pinpoint the problem.

This works well :
"""
Category.objects.get(id=6).delete()
Category.objects.get(id=7).delete()
"""
while this breaks the tree:
"""
queryset = Category.objects.filter(id__in=[6,7])
for category in queryset:
   category .delete()
"""

I understand how this can happen (the mptt fields of c7 change in the database after c6 is deleted but not in the object instance, and the delete() method considers the object's instance value probably for optimisations purposes). 

But this bug is very surprising, especially since deleting each instance individually calls the instance's delete() method (as opposed to calling the manager's delete method `Category.objects.filter(id__in=[6,7]).delete()` where breaking the tree can be expected as instance's delete() methods aren't called).

I'll open a PR with a fix and a regression test.

**Comment by matthiask** [comment]
Hopefully, this has been fixed by merging #580.
